google/cloud/spanner/client.cc

```diff
@@ -176,14 +177,25 @@ StatusOr<CommitResult> Client::Commit(
         return status;
       }
     }
-    // A transient failure (i.e., kAborted), so consider rerunning.
+    // A transient failure (e.g., kAborted), so consider rerunning.
     if (!rerun_policy->OnFailure(status)) {
       return status;  // reruns exhausted
     }
+    if (internal::IsSessionNotFound(status)) {
+      // Marks the session bad and creates a new Transaction for the next loop.
+      internal::Visit(txn, [](internal::SessionHolder& s,
+                              google::spanner::v1::TransactionSelector const&,
+                              std::int64_t) {
+        s->set_bad();
+        return true;
+      });
+      txn = MakeReadWriteTransaction();
+    } else {
+      // Create a new transaction for the next loop, but share lock priority
+      // so that we have a slightly better chance of avoiding another abort.
```

Comment:
Now that we've mentioned "session" and used a `SessionHolder` above, we can probably amend this comment to something like:
```
// Create a new transaction for the next loop, but reuse the session
// so that we have a slightly better chance of avoiding another abort.
```
That makes things a little clearer.

Reply:
Done.